ecl/hql/hqlgram.y

```diff
@@ -7976,27 +7976,29 @@ simpleDataSet
                             $$.setExpr(createDataset(no_choosesets, $3.getExpr(), $5.getExpr()));
                             $$.setPosition($1);
                         }
-    | DEDUP '(' startTopLeftRightSeqFilter optDedupFlags ')' endTopLeftRightFilter endSelectorSequence
+    | DEDUP '(' startSortOrder startTopLeftRightSeqFilter optDedupFlags ')' endTopLeftRightFilter endSelectorSequence endSortOrder
                         {
-                            IHqlExpression *ds = $3.getExpr();
-                            parser->expandWholeAndExcept(ds, $4);
+                            IHqlExpression *ds = $4.getExpr();
+                            parser->expandWholeAndExcept(ds, $5);
 
-                            IHqlExpression *flags = $4.getExpr();
-                            //checkDedup(ds, flags, $3);
-                            OwnedHqlExpr dedup = createDataset(no_dedup, ds, createComma(flags, $7.getExpr()));
+                            IHqlExpression *flags = $5.getExpr();
+                            //checkDedup(ds, flags, $4);
+                            OwnedHqlExpr dedup = createDataset(no_dedup, ds, createComma(flags, $8.getExpr()));
 
-                            parser->checkDistribution($3, dedup, false);
+                            parser->checkDistribution($4, dedup, false);
                             bool hasHash = dedup->hasAttribute(hashAtom);
                             if (hasHash || dedup->hasAttribute(allAtom))
                             {
                                 IHqlExpression * keep = dedup->queryAttribute(keepAtom);
                                 if (keep && !matchesConstantValue(keep->queryChild(0), 1))
-                                    parser->reportError(ERR_DEDUP_ALL_KEEP, $4, "KEEP is not supported for DEDUP(ALL)");
+                                    parser->reportError(ERR_DEDUP_ALL_KEEP, $5, "KEEP is not supported for DEDUP(ALL)");
                             }
                             if (hasHash)
                             {
                                 if (dedup->hasAttribute(rightAtom))
-                                    parser->reportError(ERR_DEDUP_ALL_KEEP, $4, "RIGHT is not supported for DEDUP(HASH)");
+                                    parser->reportError(ERR_DEDUP_ALL_KEEP, $5, "RIGHT is not supported for DEDUP(HASH)");
+                                if (dedup->hasAttribute(bestAtom))
+                                    parser->reportError(ERR_DEDUP_ALL_KEEP, $5, "BEST is not supported for DEDUP(HASH)");
```

Comment:
It could be supported.., albeit a slightly different implementation.
At the point it currently looks up and throws away if already in the HT, it could compare..